typed-racket-lib/typed-racket/private/parse-type.rkt

```diff
@@ -958,19 +958,35 @@
               (parse-type #'rng)
               : (-PS (attribute latent.positive) (attribute latent.negative))
               : (attribute latent.object)))]
+      ;; like ->* below but w/ a #:rest-pat present
       [(:->*^ (~var mand (->*-args #t))
               (~optional (~var opt (->*-args #f))
                          #:defaults ([opt.doms null] [opt.kws null]))
-              rest:->*-rest
+              #:rest* (rest-types-stx:non-keyword-ty ...)
```

Comment:
Are there any tests/docs for this new form?

Reply:
no -- wanted feedback on what it should actually look like for user syntax first.

I don't want to just re-use `#:rest` because I worry that will lead to ambiguous parsing, but I don't have strong feelings on what to use instead. `rest*` as in Kleene-star seems sensible to me. I dunno.

Reply:
`#:rest*` seems fine, or `#:rest-sequence` or perhaps something else?